ambry-frontend/src/main/java/com/github/ambry/frontend/FrontendRestRequestService.java

```diff
@@ -315,15 +315,14 @@ public void handleGet(final RestRequest restRequest, final RestResponseChannel r
       } else if (requestPath.matchesOperation(Operations.STATS_REPORT)) {
         getStatsReportHandler.handle(restRequest, restResponseChannel,
             (result, exception) -> submitResponse(restRequest, restResponseChannel, result, exception));
+      } else if (S3ListHandler.isListBucketRequest(restRequest) || S3MultipartUploadHandler.isListPartRequest(
+          restRequest)) {
+        s3ListHandler.handle(restRequest, restResponseChannel,
+            (result, exception) -> submitResponse(restRequest, restResponseChannel, result, exception));
```

Comment:
I suggest to have a method like `isS3Request` to just check if the request is an s3 request regardless of sub types. In this case, we can have logic like this
```
else if (S3BaseHandler.isS3Request(restRequest)) {
    if (isListBucketRequest(restRequest)) {
    } else if (isMultipartListRequest(restRequest)) {
    } else {
        // throw error
    }
}
```